### PR TITLE
Add AppWorkload.spec.buildRef

### PR DESCRIPTION
**Pull request by matt-royal**
## Is there a related GitHub Issue?
 #1525 

## What is this change about?
See issue

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See issue

### CODE REVIEW

File: controllers/controllers/workloads/integration/cfprocess_controller_integration_test.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var _ = Describe("CFProcessReconciler Integration Tests", func() {
+var _ = Describe("CFProcessReconciler Integration Tests",  func() {
```
Comment (tcdowney) [inline controllers/controllers/workloads/integration/cfprocess_controller_integration_test.go:20]: Nit: extra whitespace

File: controllers/controllers/workloads/integration/cfprocess_controller_integration_test.go [hunk 2]
```diff
@@ -145,25 +145,26 @@ var _ = Describe("CFProcessReconciler Integration Tests", func() {
...
-				g.Expect(appWorkload.Spec.DiskMiB).To(Equal(cfProcess.Spec.DiskQuotaMB), "appWorkload DiskMB does not match")
-				g.Expect(appWorkload.Spec.MemoryMiB).To(Equal(cfProcess.Spec.MemoryMB), "appWorkload MemoryMB does not match")
-				g.Expect(appWorkload.Spec.Image).To(Equal(cfBuild.Status.Droplet.Registry.Image), "appWorkload Image does not match Droplet")
-				g.Expect(appWorkload.Spec.ImagePullSecrets).To(Equal(cfBuild.Status.Droplet.Registry.ImagePullSecrets), "appWorkload ImagePullSecrets does not match Droplet")
-				g.Expect(appWorkload.Spec.ProcessType).To(Equal(processTypeWeb), "appWorkload process type does not match")
-				g.Expect(appWorkload.Spec.AppGUID).To(Equal(cfApp.Name), "appWorkload app GUID does not match CFApp")
-				g.Expect(appWorkload.Spec.Ports).To(Equal(cfProcess.Spec.Ports), "appWorkload ports do not match")
-				g.Expect(appWorkload.Spec.Instances).To(Equal(int32(cfProcess.Spec.DesiredInstances)), "appWorkload desired instances does not match CFApp")
-				g.Expect(appWorkload.Spec.CPUMillicores).To(Equal(int64(100)), "expected cpu request to be 100m (Based on 1024MiB memory)")
```
Comment (tcdowney) [inline controllers/controllers/workloads/integration/cfprocess_controller_integration_test.go:166]: > "expected cpu request to be 100m (Based on 1024MiB memory)"

I think some of these failure annotations might be helpful. It feels like they taking the place of discrete `It` descriptions in some cases.

Like in this instance `100` is kind of a magic number and a description like "It scales CPU request based on memory" could be useful context to someone trying to debug a test failure here.

I agree that most of these aren't very useful, though.

**Review by tcdowney — COMMENTED** [review]
Overall LGTM, added a couple minor suggestions.

File: controllers/controllers/workloads/integration/cfprocess_controller_integration_test.go (hunk 2 above)
Comment (matt-royal) [reply controllers/controllers/workloads/integration/cfprocess_controller_integration_test.go:166]: Good point. I just deleted them indiscriminately. I'll take a pass and add some back.

**Review by tcdowney — APPROVED** [review]

**Pull request merged**